### DISCUSSION TITLE
Add additional file types to New File dialog

### DISCUSSION THREAD

**Discussion by profbbrown** [eclipse-theia/theia#13339] [Q&A]
By default, the New File pop-up dialog displays "Text File" and "Jupyter Notebook File". How can I add some additional file types, such as "C File", "Java File", or "Makefile"?

I think I need to write an extension, but I've never done it before. A pointer in the right direction would be much appreciated.

**Comment by msujew** [comment]
Hey @profbbrown,

you can either write a Theia extension or build a vscode plugin. Either way, you need to add new elements to a specific menu path, see:

https://github.com/eclipse-theia/theia/blob/b15b639fa3b64a9518efdf3e82ea9098915ba145/packages/core/src/browser/common-frontend-contribution.ts#L1422-L1454
